src/components/ComponentPages/ProfileInfo/index.tsx

```diff
@@ -124,6 +128,65 @@ const ProfileInfo = () => {
       }
     }
   };
+  const handleAddressChange = (value) => {
+    setAddress(value)
+  }
+
+  const handleAddressSelect = async (address, placeId) => {
+    setAddress(address)
+    const results = await geocodeByAddress(address);
+    const [place] = await geocodeByPlaceId(placeId);
+    const { long_name: postalCode = '' } =
+      place.address_components.find(c => c.types.includes('postal_code')) || {};
+    var city = ""
+    var country = ""
+    var address2 = ""
+
+    for (const component of results[0].address_components) {
+      const componentType = component.types[0];
+      switch (componentType) {
+        case "neighborhood": {
+          address2 = component.short_name
+          break;
+        }
+        case "political": {
+          address2 = address2 + ", " + component.long_name;
+          break;
+        }
+        case "neighborhood": {
+          address2 = address2 + ", " + component.long_name;
+          break;
+        }
+        case "sublocality_level_2": {
+          address2 = address2 + ", " + component.long_name;
+          break;
+        }
+        case "sublocality_level_1": {
+          address2 = address2 + ", " + component.long_name;
+          break;
+        }
+        case "locality": {
+          city = component.long_name;
+          break;
+        }
+        case "administrative_area_level_1": {
+          var state = component.long_name;
+          break;
+        }
+        case "country": {
+          country = component.long_name;
+          break;
+        }
+      }
+    }
+    address2 = address2.replace(/^,|,$/g, '');
+
+    form.setFieldsValue({ ["address_2"]: address2 })
+    form.setFieldsValue({ ["postal_code"]: postalCode })
+    form.setFieldsValue({ ["city"]: city })
+    form.setFieldsValue({ ["state"]: state })
+    form.setFieldsValue({ ["country"]: country })
+  }
```

Comment:
Probably we need to break this. This doesn't looks good

```javascript

for (const component of results[0].address_components) {
      const componentType = component.types[0];
      
      address2 = getAddress(component, address2);

 .....
......
.....
}


// Create a function similar to this
function getAddress( type, address ) {
    if ( type === "neighbourhood" ) {
        return address
    }
    else if ( type.match( /$neighborhood^|$sublocality_level_2^|$sublocality_level_1^/ ) ) {
        return address + ", " + type.long_name
    }
    else { 
        return 0
    }
  }

```

Reply:
`city`, `state` and `country` are state forward

Reply:
done